lib/ansible/galaxy/collection.py

```diff
@@ -0,0 +1,385 @@
+# Copyright: (c) 2019, Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+import fnmatch
+import json
+import os
+import shutil
+import tarfile
+import tempfile
+import time
+import uuid
+import yaml
+
+from contextlib import contextmanager
+from hashlib import sha256
+from io import BytesIO
+from yaml.error import YAMLError
+
+import ansible.constants as C
+import ansible.module_utils.six.moves.urllib.error as urllib_error
+from ansible.errors import AnsibleError
+from ansible.module_utils._text import to_bytes, to_native, to_text
+from ansible.utils.display import Display
+from ansible.utils.hashing import secure_hash, secure_hash_s
+
+from ansible.module_utils.urls import open_url
+
+
+display = Display()
+
+MANIFEST_FORMAT = 1
+
+
+def build_collection(collection_path, output_path, force):
+    """
+    Creates the Ansible collection artifact in a .tar.gz file.
+
+    :param collection_path: The path to the collection to build. This should be the directory that contains the
+        galaxy.yml file.
+    :param output_path: The path to create the collection build artifact. This should be a directory.
+    :param force: Whether to overwrite an existing collection build artifact or fail.
+    :return: The path to the collection build artifact.
+    """
+    galaxy_path = os.path.join(collection_path, 'galaxy.yml')
+    if not os.path.exists(galaxy_path):
+        raise AnsibleError("The collection galaxy.yml path '%s' does not exist." % galaxy_path)
+
+    collection_meta = _get_galaxy_yml(galaxy_path)
+    file_manifest = _build_files_manifest(collection_path)
+    collection_manifest = _build_manifest(**collection_meta)
+
+    collection_output = os.path.join(output_path, "%s-%s-%s.tar.gz" % (collection_meta['namespace'],
+                                                                       collection_meta['name'],
+                                                                       collection_meta['version']))
+
+    if os.path.exists(collection_output):
```

Comment:
If you turn this into
```suggestion
    if not os.path.exists(collection_output):
    	_build_collection_tar(
    		collection_path, collection_output, collection_manifest, file_manifest,
    	)
```
you could reduce some nesting which improves readability.